Index: Engine/Shaders/PostProcessAmbient.usf
===================================================================
--- Engine/Shaders/PostProcessAmbient.usf	(revision 169)
+++ Engine/Shaders/PostProcessAmbient.usf	(revision 170)
@@ -356,7 +356,15 @@
 	float2 ScreenSpacePos = UVAndScreenPos.zw;
 	int2 PixelPos = (int2)SvPosition.xy;
 
-	float3 ScreenVector = normalize(mul(float4(ScreenSpacePos, 1, 0), View.ScreenToWorld).xyz);
+	// v6
+	float3 ScreenVectorU = mul(float4(ScreenSpacePos, 1, 0), View.ScreenToWorld).xyz;
+	//float3 ScreenVector = normalize(mul(float4(ScreenSpacePos, 1, 0), View.ScreenToWorld).xyz);
+	
+	// v6
+	float SceneDepth = CalcSceneDepth( UV );
+	float3 WorldPosition = ScreenVectorU * SceneDepth + View.WorldCameraOrigin;
+	
+	float3 ScreenVector = normalize(WorldPosition - View.WorldEyeOrigin);
 
 	float3 N = GBuffer.WorldNormal;
 	float3 V = -ScreenVector;
Index: Engine/Shaders/ForEachLight.usf
===================================================================
--- Engine/Shaders/ForEachLight.usf	(revision 169)
+++ Engine/Shaders/ForEachLight.usf	(revision 170)
@@ -17,7 +17,9 @@
 
 float3 LightingForOneLight(FMaterialPixelParameters Parameters, uint LightIndex, FScreenSpaceData ScreenSpaceData)
 {
-	float3 CameraVector = normalize(Parameters.AbsoluteWorldPosition - View.WorldCameraOrigin);
+	// v6
+	float3 CameraVector = normalize(Parameters.AbsoluteWorldPosition - View.WorldEyeOrigin);
+	// float3 CameraVector = normalize(Parameters.AbsoluteWorldPosition - View.WorldCameraOrigin);
 	
 	FDeferredLightData LightData = (FDeferredLightData)0;
 	LightData.LightPositionAndInvRadius = ForwardLightData.LightPositionAndInvRadius[LightIndex];
Index: Engine/Shaders/DeferredLightingCommon.usf
===================================================================
--- Engine/Shaders/DeferredLightingCommon.usf	(revision 169)
+++ Engine/Shaders/DeferredLightingCommon.usf	(revision 170)
@@ -234,7 +234,9 @@
 		// Also fix up the fade between dynamic and static shadows
 		// to work with plane splits rather than spheres.
 
-		float DynamicShadowFraction = DistanceFromCameraFade(ScreenSpaceData, LightData, WorldPosition, View.WorldCameraOrigin);
+		// v6
+		float DynamicShadowFraction = DistanceFromCameraFade(ScreenSpaceData, LightData, WorldPosition, View.WorldEyeOrigin);
+		// float DynamicShadowFraction = DistanceFromCameraFade(ScreenSpaceData, LightData, WorldPosition, View.WorldCameraOrigin);
 		// For a directional light, fade between static shadowing and the whole scene dynamic shadowing based on distance + per object shadows
 		OpaqueShadowTerm = lerp(LightAttenuation.x, StaticShadowing, DynamicShadowFraction);
 		// Fade between SSS dynamic shadowing and static shadowing based on distance
Index: Engine/Shaders/InstancedStereo.usf
===================================================================
--- Engine/Shaders/InstancedStereo.usf	(revision 169)
+++ Engine/Shaders/InstancedStereo.usf	(revision 170)
@@ -25,6 +25,8 @@
 	float4 InvDeviceZToWorldZTransform;
 	float4 ScreenPositionScaleBias;
 	float3 WorldCameraOrigin;
+	// v6
+	float3 WorldEyeOrigin;
 	float3 TranslatedWorldCameraOrigin;
 	float3 WorldViewOrigin;
 	float3 PreViewTranslation;
@@ -70,6 +72,8 @@
 	Result.InvDeviceZToWorldZTransform = View.InvDeviceZToWorldZTransform;
 	Result.ScreenPositionScaleBias = View.ScreenPositionScaleBias;
 	Result.WorldCameraOrigin = View.WorldCameraOrigin;
+	// v6
+	Result.WorldEyeOrigin = View.WorldEyeOrigin;
 	Result.TranslatedWorldCameraOrigin = View.TranslatedWorldCameraOrigin;
 	Result.WorldViewOrigin = View.WorldViewOrigin;
 	Result.PreViewTranslation = View.PreViewTranslation;
@@ -149,6 +153,8 @@
 		Result.InvDeviceZToWorldZTransform = View.InvDeviceZToWorldZTransform;
 		Result.ScreenPositionScaleBias = View.ScreenPositionScaleBias;
 		Result.WorldCameraOrigin = View.WorldCameraOrigin;
+		// v6
+		Result.WorldEyeOrigin = View.WorldEyeOrigin;
 		Result.TranslatedWorldCameraOrigin = View.TranslatedWorldCameraOrigin;
 		Result.WorldViewOrigin = View.WorldViewOrigin;
 		Result.PreViewTranslation = View.PreViewTranslation;
@@ -189,6 +195,8 @@
 		Result.InvDeviceZToWorldZTransform = InstancedView.InvDeviceZToWorldZTransform;
 		Result.ScreenPositionScaleBias = InstancedView.ScreenPositionScaleBias;
 		Result.WorldCameraOrigin = InstancedView.WorldCameraOrigin;
+		// v6
+		Result.WorldEyeOrigin = InstancedView.WorldEyeOrigin;
 		Result.TranslatedWorldCameraOrigin = InstancedView.TranslatedWorldCameraOrigin;
 		Result.WorldViewOrigin = InstancedView.WorldViewOrigin;
 		Result.PreViewTranslation = InstancedView.PreViewTranslation;
Index: Engine/Shaders/TiledDeferredLightShaders.usf
===================================================================
--- Engine/Shaders/TiledDeferredLightShaders.usf	(revision 169)
+++ Engine/Shaders/TiledDeferredLightShaders.usf	(revision 170)
@@ -239,7 +239,9 @@
 	// Lookup GBuffer properties once per pixel 
 	FScreenSpaceData ScreenSpaceData = GetScreenSpaceDataUint(PixelPos);
 	FGBufferData InGBufferData = ScreenSpaceData.GBuffer; 
-	float3 CameraVector = normalize(WorldPosition - View.WorldCameraOrigin);
+	// v6
+	float3 CameraVector = normalize(WorldPosition - View.WorldEyeOrigin);
+	// float3 CameraVector = normalize(WorldPosition - View.WorldCameraOrigin);
 		
 	BRANCH
 	if (InGBufferData.ShadingModelID != SHADINGMODELID_UNLIT)
Index: Engine/Shaders/LightFunctionPixelShader.usf
===================================================================
--- Engine/Shaders/LightFunctionPixelShader.usf	(revision 169)
+++ Engine/Shaders/LightFunctionPixelShader.usf	(revision 170)
@@ -35,7 +35,9 @@
 	float3 AbsoluteWorldPosition = SvPositionToWorld(SvPosition);
 
 	// Calculate radial view distance for stable fading
-	float ViewDistance = length(View.WorldCameraOrigin - AbsoluteWorldPosition);
+	// v6
+	float ViewDistance = length(View.WorldEyeOrigin - AbsoluteWorldPosition);
+	//float ViewDistance = length(View.WorldCameraOrigin - AbsoluteWorldPosition);
 
 	float GreyScale;
 	{
Index: Engine/Shaders/MaterialTemplate.usf
===================================================================
--- Engine/Shaders/MaterialTemplate.usf	(revision 169)
+++ Engine/Shaders/MaterialTemplate.usf	(revision 170)
@@ -676,9 +676,13 @@
 {
 #if MATERIAL_ATMOSPHERIC_FOG
 	// WorldPosition default value is Parameters.AbsoluteWorldPosition if not overridden by the user
-	float3 ViewVector = AbsoluteWorldPosition - ResolvedView.WorldCameraOrigin;
+	// v6
+	float3 ViewVector = AbsoluteWorldPosition - ResolvedView.WorldEyeOrigin;
+	// float3 ViewVector = AbsoluteWorldPosition - ResolvedView.WorldCameraOrigin;
 	float SceneDepth = length(ViewVector);
-	return GetAtmosphericFog(ResolvedView.WorldCameraOrigin, ViewVector, SceneDepth, float3(0.f, 0.f, 0.f));
+	// v6
+	return GetAtmosphericFog(ResolvedView.WorldEyeOrigin, ViewVector, SceneDepth, float3(0.f, 0.f, 0.f));
+	// return GetAtmosphericFog(ResolvedView.WorldCameraOrigin, ViewVector, SceneDepth, float3(0.f, 0.f, 0.f));
 #else
 	return float4(0.f, 0.f, 0.f, 0.f);
 #endif
Index: Engine/Shaders/DeferredDecal.usf
===================================================================
--- Engine/Shaders/DeferredDecal.usf	(revision 169)
+++ Engine/Shaders/DeferredDecal.usf	(revision 170)
@@ -79,7 +79,9 @@
 	MaterialParameters.WorldPosition_CamRelative = 
 	MaterialParameters.WorldPosition_NoOffsets = 
 	MaterialParameters.WorldPosition_NoOffsets_CamRelative = mul(DecalToWorld, float4(Pos, 1)).xyz;
-	MaterialParameters.CameraVector = normalize(View.WorldCameraOrigin - MaterialParameters.AbsoluteWorldPosition);
+	// v6
+	MaterialParameters.CameraVector = normalize(View.WorldEyeOrigin - MaterialParameters.AbsoluteWorldPosition);
+	//MaterialParameters.CameraVector = normalize(View.WorldCameraOrigin - MaterialParameters.AbsoluteWorldPosition);
 
 	FPixelMaterialInputs PixelMaterialInputs;
 	CalcPixelMaterialInputs(MaterialParameters, PixelMaterialInputs);
@@ -217,7 +219,9 @@
 
 		float4 SvPositionInDistance = float4(SvPosition.xy, ConvertToDeviceZ(SceneW), 1);
 
-		float3 WSStartRay = View.WorldCameraOrigin;
+		// v6
+		float3 WSStartRay = View.WorldEyeOrigin;
+		// float3 WSStartRay = View.WorldCameraOrigin;
 		float3 WSEndRay = SvPositionToWorld(SvPositionInDistance);
 
 		// in object/decal space
@@ -258,7 +262,9 @@
 		WSPosition = SvPositionToWorld(SvPosition);
 	}
 
-	float3 CameraVector = normalize(View.WorldCameraOrigin - WSPosition);
+	// v6
+	float3 CameraVector = normalize(View.WorldEyeOrigin - WSPosition);
+	// float3 CameraVector = normalize(View.WorldCameraOrigin - WSPosition);
 
 	// can be optimized
 	float3 DecalVector = OSPosition * 0.5f + 0.5f;
Index: Engine/Shaders/DeferredLightPixelShaders.usf
===================================================================
--- Engine/Shaders/DeferredLightPixelShaders.usf	(revision 169)
+++ Engine/Shaders/DeferredLightPixelShaders.usf	(revision 170)
@@ -69,6 +69,9 @@
 		float SceneDepth = CalcSceneDepth(InUV);
 		float3 WorldPosition = ScreenVector * SceneDepth + View.WorldCameraOrigin;
 
+		// v6
+		CameraVector = normalize(WorldPosition - View.WorldEyeOrigin);
+
 		FDeferredLightData LightData = SetupLightDataForStandardDeferred();
 		
 		uint2 Random = ScrambleTEA( uint2( SVPos.xy ) );
@@ -103,7 +106,9 @@
 		float SceneDepth = CalcSceneDepth(ScreenUV);
 		float4 HomogeneousWorldPosition = mul(float4(InScreenPosition.xy / InScreenPosition.w * SceneDepth, SceneDepth, 1), View.ScreenToWorld);
 		float3 WorldPosition = HomogeneousWorldPosition.xyz / HomogeneousWorldPosition.w;
-		float3 CameraVector = normalize(WorldPosition - View.WorldCameraOrigin);
+		// v6
+		float3 CameraVector = normalize(WorldPosition - View.WorldEyeOrigin);
+		// float3 CameraVector = normalize(WorldPosition - View.WorldCameraOrigin);
 
 		FDeferredLightData LightData = SetupLightDataForStandardDeferred();
 
Index: Engine/Shaders/ReflectionEnvironmentComputeShaders.usf
===================================================================
--- Engine/Shaders/ReflectionEnvironmentComputeShaders.usf	(revision 169)
+++ Engine/Shaders/ReflectionEnvironmentComputeShaders.usf	(revision 170)
@@ -607,7 +607,9 @@
 	
 	float4 HomogeneousWorldPosition = mul(float4(ScreenPosition * SceneDepth, SceneDepth, 1), View.ScreenToWorld);
 	float3 WorldPosition = HomogeneousWorldPosition.xyz / HomogeneousWorldPosition.w;
-	float3 CameraToPixel = normalize(WorldPosition - View.WorldCameraOrigin);
+	// v6
+	float3 CameraToPixel = normalize(WorldPosition - View.WorldEyeOrigin);
+	// float3 CameraToPixel = normalize(WorldPosition - View.WorldCameraOrigin);
 	float3 ReflectionVector = reflect(CameraToPixel, GBuffer.WorldNormal);
 	float IndirectIrradiance = GBuffer.IndirectIrradiance;
 	
Index: Engine/Shaders/ReflectionEnvironmentShaders.usf
===================================================================
--- Engine/Shaders/ReflectionEnvironmentShaders.usf	(revision 169)
+++ Engine/Shaders/ReflectionEnvironmentShaders.usf	(revision 170)
@@ -64,7 +64,9 @@
 	}
 	else if (SkyLightCaptureParameters.x == 1)
 	{
-		float RadialDistance = length(WorldPosition - View.WorldCameraOrigin);
+		// v6
+		float RadialDistance = length(WorldPosition - View.WorldEyeOrigin);
+		// float RadialDistance = length(WorldPosition - View.WorldCameraOrigin);
 		float MaxDistance = SkyLightCaptureParameters.y;
 
 		// Setup alpha to fade out smoothly past the max distance
@@ -351,7 +353,9 @@
 		float SceneDepth = CalcSceneDepth(ScreenUV);
 		float4 HomogeneousWorldPosition = mul(float4(InScreenPosition.xy / InScreenPosition.w * SceneDepth, SceneDepth, 1), View.ScreenToWorld);
 		float3 WorldPosition = HomogeneousWorldPosition.xyz / HomogeneousWorldPosition.w;
-		float3 CameraToPixel = normalize(WorldPosition - View.WorldCameraOrigin);
+		// v6
+		float3 CameraToPixel = normalize(WorldPosition - View.WorldEyeOrigin);
+		// float3 CameraToPixel = normalize(WorldPosition - View.WorldCameraOrigin);
 		float3 ReflectionVector = reflect(CameraToPixel, InGBufferData.WorldNormal);
 
 		float3 ProjectedCaptureVector = ReflectionVector;
Index: Engine/Shaders/AtmosphericFogShader.usf
===================================================================
--- Engine/Shaders/AtmosphericFogShader.usf	(revision 169)
+++ Engine/Shaders/AtmosphericFogShader.usf	(revision 170)
@@ -57,7 +57,9 @@
 	float4 ScreenVector : TEXCOORD1,
 	out float4 OutColor : SV_Target0)
 {
-	OutColor = GetAtmosphericFog(View.WorldCameraOrigin, ScreenVector.xyz, CalcSceneDepth(TexCoord), float3(0, 0, 0));
+	// v6
+	OutColor = GetAtmosphericFog(View.WorldEyeOrigin, ScreenVector.xyz, CalcSceneDepth(TexCoord), float3(0, 0, 0));
+	// OutColor = GetAtmosphericFog(View.WorldCameraOrigin, ScreenVector.xyz, CalcSceneDepth(TexCoord), float3(0, 0, 0));
 #if !ATMOSPHERIC_NO_LIGHT_SHAFT
 	float LightShaftMask = Texture2DSample(OcclusionTexture, OcclusionTextureSampler, TexCoord).x;
 	OutColor.rgb = OutColor.rgb * LightShaftMask;
Index: Engine/Shaders/TranslucentLightInjectionShaders.usf
===================================================================
--- Engine/Shaders/TranslucentLightInjectionShaders.usf	(revision 169)
+++ Engine/Shaders/TranslucentLightInjectionShaders.usf	(revision 170)
@@ -204,7 +204,9 @@
 
 		// fade shadows in the distance
 		#if !RADIAL_ATTENUATION 
-			float Depth = dot(WorldPositionForLighting - View.WorldCameraOrigin, View.ViewForward);
+			// v6
+			float Depth = dot(WorldPositionForLighting - View.WorldEyeOrigin, View.ViewForward);
+			// float Depth = dot(WorldPositionForLighting - View.WorldCameraOrigin, View.ViewForward);
 			float DistanceFade = saturate(Depth * ShadowInjectParams.z + ShadowInjectParams.w);
 			DynamicShadowFactor = lerp(DynamicShadowFactor, 1.0f, DistanceFade * DistanceFade);
 		#endif
Index: Engine/Source/Runtime/Engine/Public/SceneView.h
===================================================================
--- Engine/Source/Runtime/Engine/Public/SceneView.h	(revision 169)
+++ Engine/Source/Runtime/Engine/Public/SceneView.h	(revision 170)
@@ -75,6 +75,9 @@
 	const AActor* ViewActor;
 	FViewElementDrawer* ViewElementDrawer;
 
+	// v6
+	FVector ViewOriginForLighting;
+
 	FLinearColor BackgroundColor;
 	FLinearColor OverlayColor;
 	FLinearColor ColorScale;
@@ -120,6 +123,7 @@
 		, SceneViewStateInterface(NULL)
 		, ViewActor(NULL)
 		, ViewElementDrawer(NULL)
+		, ViewOriginForLighting( FLT_MAX )
 		, BackgroundColor(FLinearColor::Transparent)
 		, OverlayColor(FLinearColor::Transparent)
 		, ColorScale(FLinearColor::White)
@@ -157,6 +161,8 @@
 		PreShadowTranslation = FVector::ZeroVector;
 		PreViewTranslation = FVector::ZeroVector;
 		ViewOrigin = FVector::ZeroVector;
+		// v6
+		ViewOriginForLighting = FVector( FLT_MAX );
 		ProjectionScale = FVector2D::ZeroVector;
 		TemporalAAProjJitter = FVector2D::ZeroVector;
 		ScreenScale = 1.f;
@@ -180,6 +186,8 @@
 	FVector		PreViewTranslation;
 	/** To support ortho and other modes this is redundant, in world space */
 	FVector		ViewOrigin;
+	// v6
+	FVector		ViewOriginForLighting;
 	/** Scale applied by the projection matrix in X and Y. */
 	FVector2D	ProjectionScale;
 	/** TemporalAA jitter offset currently stored in the projection matrix */
@@ -384,6 +392,8 @@
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector4, InvDeviceZToWorldZTransform)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER_EX(FVector4, ScreenPositionScaleBias, EShaderPrecisionModifier::Half)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldCameraOrigin)
+	// v6
+	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldEyeOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, TranslatedWorldCameraOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldViewOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, PreViewTranslation)
@@ -425,6 +435,8 @@
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector4, InvDeviceZToWorldZTransform)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER_EX(FVector4, ScreenPositionScaleBias, EShaderPrecisionModifier::Half)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldCameraOrigin)
+	// v6
+	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldEyeOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, TranslatedWorldCameraOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, WorldViewOrigin)
 	DECLARE_UNIFORM_BUFFER_STRUCT_MEMBER(FVector, PreViewTranslation)
Index: Engine/Source/Runtime/Engine/Private/SceneView.cpp
===================================================================
--- Engine/Source/Runtime/Engine/Private/SceneView.cpp	(revision 169)
+++ Engine/Source/Runtime/Engine/Private/SceneView.cpp	(revision 170)
@@ -417,6 +417,7 @@
 	if(IsPerspectiveProjection())
 	{
 		ViewMatrices.ViewOrigin = ViewOrigin;
+		ViewMatrices.ViewOriginForLighting = InitOptions.ViewOriginForLighting;
 	}
 #if WITH_EDITOR
 	else if (InitOptions.bUseFauxOrthoViewPos)
Index: Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderingPolicy.cpp
===================================================================
--- Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderingPolicy.cpp	(revision 169)
+++ Engine/Source/Runtime/SlateRHIRenderer/Private/SlateRHIRenderingPolicy.cpp	(revision 170)
@@ -220,6 +220,7 @@
 	ViewUniformShaderParameters.InvDeviceZToWorldZTransform = View->InvDeviceZToWorldZTransform;
 	ViewUniformShaderParameters.WorldViewOrigin = View->ViewMatrices.ViewOrigin;
 	ViewUniformShaderParameters.WorldCameraOrigin = View->ViewMatrices.ViewOrigin;
+	ViewUniformShaderParameters.WorldEyeOrigin = View->ViewMatrices.ViewOrigin;
 	ViewUniformShaderParameters.TranslatedWorldCameraOrigin = ViewUniformShaderParameters.WorldCameraOrigin + View->ViewMatrices.PreViewTranslation;
 	ViewUniformShaderParameters.PreViewTranslation = View->ViewMatrices.PreViewTranslation;
 	ViewUniformShaderParameters.ScreenPositionScaleBias = ScreenPositionScaleBias;
Index: Engine/Source/Runtime/Renderer/Private/SceneRendering.cpp
===================================================================
--- Engine/Source/Runtime/Renderer/Private/SceneRendering.cpp	(revision 169)
+++ Engine/Source/Runtime/Renderer/Private/SceneRendering.cpp	(revision 170)
@@ -512,6 +512,8 @@
 	ViewUniformShaderParameters.InvDeviceZToWorldZTransform = InvDeviceZToWorldZTransform;
 	ViewUniformShaderParameters.WorldViewOrigin = EffectiveViewToTranslatedWorld.TransformPosition(FVector(0)) - ViewMatrices.PreViewTranslation;
 	ViewUniformShaderParameters.WorldCameraOrigin = ViewMatrices.ViewOrigin;
+	// v6
+	ViewUniformShaderParameters.WorldEyeOrigin = ViewMatrices.ViewOriginForLighting != FVector( FLT_MAX ) ? ViewMatrices.ViewOriginForLighting : ViewMatrices.ViewOrigin;
 	ViewUniformShaderParameters.TranslatedWorldCameraOrigin = ViewMatrices.ViewOrigin + ViewMatrices.PreViewTranslation;
 	ViewUniformShaderParameters.PreViewTranslation = ViewMatrices.PreViewTranslation;
 	ViewUniformShaderParameters.PrevProjection = PrevViewMatrices.ProjMatrix;
